src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx

```diff
@@ -14,32 +14,58 @@ import {
 } from '@material-ui/core';
 import { withStyles } from '@material-ui/core/styles';
 import { Add, Edit } from '@material-ui/icons';
-import PropTypes from 'prop-types';
-import { addCustomEvent, editCustomEvent } from '../../../actions/AppStoreActions';
+import { addCustomEvent, editCustomEvent } from '../../../../actions/AppStoreActions';
 import ScheduleSelector from './ScheduleSelector';
 import ReactGA from 'react-ga';
-import analyticsEnum, { logAnalytics } from '../../../analytics';
-import { isDarkMode } from '../../../helpers';
+import analyticsEnum, { logAnalytics } from '../../../../analytics';
+import { isDarkMode } from '../../../../helpers';
 
-const styles = () => ({
-    container: {
-        display: 'flex',
-        flexWrap: 'wrap',
-    },
+const styles = {
     textField: {
         minWidth: 120,
     },
-});
+};
+
+export interface CustomEvent {
+    title: string,
+    start: string,
+    end: string,
+    days: boolean[],
+    scheduleIndices: number[],
+    customEventID: number,
+    color?: string
+}
 
-class CustomEventDialog extends PureComponent {
-    state = {
+interface CustomEventDialogProps {
+    customEvent?: CustomEvent,
+    onDialogClose?: ()=>void,
+    currentScheduleIndex: number,
+    scheduleNames: any[] // TODO: statically type this.
+};
+
+interface CustomEventDialogState {
+    open: boolean,// the properties under this are copies of CustomEvent properties.
+    title: string,
+    start: string,
+    end: string,
+    days: boolean[],
+    scheduleIndices: number[],
+    customEventID: number
+}
```

Comment:
Can we just extend CustomEvent?
```suggestion
interface CustomEventDialogState extends CustomEvent {
    open: boolean
}
```

Reply:
good idea